src/contexts/IntlContext.tsx

```diff
@@ -21,9 +21,13 @@ interface LanguageName {
   englishName: string;
 }
 
+export type SupportedLanguageName = Omit<LanguageName, "locale"> & {
```

Comment:
For the sake of the language selector, I created this subset of `LangugageName` so that the language selector knows that only selected languages are being used. Otherwise typescript thinks `supportedLanguages` is all `LanguageNames`, which is not compatible with `setAppLocale` which only takes `SupportedLanguageName[locale]`